extension/goplugin/database.go

```diff
@@ -64,7 +64,11 @@ func (db *Database) Begin() (goext.ITransaction, error) {
 // BeginTx starts a new transaction with options
 func (db *Database) BeginTx(ctx goext.Context, options *goext.TxOptions) (goext.ITransaction, error) {
 	opts := transaction.TxOptions{IsolationLevel: transaction.Type(options.IsolationLevel)}
-	t, err := db.raw.BeginTx(context.Background(), &opts)
+	newTimedContext := ctx.GetContext()
+	if _, ok := newTimedContext.Deadline();!ok{
+		newTimedContext = context.WithTimeout(newTimedContext,5* time.Second)
```

Comment:
please run goimports to format the code before pushing it. in GoLand you could add it in File->Tools->File Watchers - it will run whenever you save a file